iati/core/tests/test_schemas.py

```diff
@@ -2,56 +2,79 @@
 
-        Returns:
-            iati.core.Schema: A Schema that has been initialised with basic values.
+    """
+    schema_name = iati.core.tests.utilities.SCHEMA_ACTIVITY_NAME_VALID
 
-        """
-        schema_name = iati.core.tests.utilities.SCHEMA_NAME_VALID
+    return iati.core.default.schema(schema_name)
 
-        return iati.core.Schema(name=schema_name)
 
-    def test_schema_default_attributes(self):
-        """Check a Schema's default attributes are correct."""
-        schema = iati.core.Schema()
+def default_organisation_schema():
+    """Create a very basic organisaion schema.
 
-        assert schema.name is None
+    Returns:
+        iati.core.OrganisaionSchema: An OrganisaionSchema that has been initialised based on the default IATI Organisaion Schema.
 
-    @pytest.mark.parametrize("invalid_name", iati.core.tests.utilities.find_parameter_by_type(['str', 'none'], False))
-    def test_schema_name_instance(self, invalid_name):
-        """Check that an Error is raised when attempting to load a Schema that does not exist.
+    """
+    schema_name = iati.core.tests.utilities.SCHEMA_ORGANISATION_NAME_VALID
+
+    return iati.core.default.schema(schema_name)
 
-        Todo:
-            Check for type errors when the type is incorrect.
+
+class TestSchemas(object):
+    """A container for tests relating to Schemas."""
+
+    @pytest.fixture(params=[
+        iati.core.tests.utilities.SCHEMA_ACTIVITY_NAME_VALID,
+        iati.core.tests.utilities.SCHEMA_ORGANISATION_NAME_VALID
+    ])
+    def schemas_initialised(self, request):
+        """Create both an ActivitySchema and OrganisaionSchema class.
+        For use where both ActivitySchema and OrganisaionSchema must produce the same result.
+
+        Returns:
+            iati.core.ActivitySchema / iati.core.OrganisaionSchema: An activity and organisaion that has been initialised based on the default IATI Activity and Organisaion schemas.
 
         """
-        with pytest.raises(TypeError) as excinfo:
-            iati.core.Schema(invalid_name)
+        schema_name = request.param
 
-        assert 'The name of the Schema is an invalid type. Must be a string, though was a' in str(excinfo.value)
+        return iati.core.default.schema(schema_name)
 
-    def test_schema_define_from_xsd(self, schema_initialised):
+    @pytest.mark.parametrize("schema_type, expected_value", [
+        (default_activity_schema, 'iati-activities'),
+        (default_organisation_schema, 'iati-organisations')
+    ])
+    def test_schema_default_attributes(self, schema_type, expected_value):
+        """Check a Schema's default attributes are correct."""
+        schema = schema_type()
+
+        assert schema.root_element_name == expected_value
+
+    def test_schema_define_from_xsd(self, schemas_initialised):
         """Check that a Schema can be generated from an XSD definition."""
-        schema = schema_initialised
+        schema = schemas_initialised
```

Comment:
Redundant assignment

(there are a number of cases throughout where there is an assignment simply to change a variable name)